### DISCUSSION TITLE
A-B-C mining asset Values.

### DISCUSSION THREAD

**Discussion by puchungotv** [tearust/teaproject#19] [Q&A] [answered]
How is the rarity of the mining asset determined?

Is there a way to also guarantee the quality of the asset?

**Comment by kevingzhang** [comment, marked as answer]
> How is the rarity of the mining asset determined?

In epoch 1.xx and 2.xx, every contestant will have the same number of coupon that redeemed into CML. The number of CML is fixed number. This is where the scarcity comes from. In the real main net, the DAO will issue new CML seeds based on an algorithm to make sure the CML asset is always scared.

**Comment by kevingzhang** [comment]
> Is there a way to also guarantee the quality of the asset?
I am not sure if I understood your question. Can you explain in detail?

**Reply by puchungotv** [reply]
From the Mining video i understood that the quality of the seed (abc) was random but is actually fixed.
So that answered my question. 

Thanks.

**Reply by kevingzhang** [reply]
No, they are not fixed, they are random. You can see the performance value is different for every CML.
They are generated in the genesis block randomly. But which seeds belonging to you is random. You will know when you redeem your coupon. So it is randomness.

**Reply by puchungotv** [reply]
Got it.

In real main net,  that will also be the case?. All new CMLs will have a random quality seed?.  making B the most common and A rare?.

**Reply by kevingzhang** [reply]
In the real main net, it will be more complicated. There will be more types, not only A/B/C. Please read my old blog https://pushbar.medium.com/camillia-an-nft-that-could-die-6ed535cccce

The DAO will generate new seeds depends on many factors, it will NOT maintain the A/B/C = 1:2:4. It will depend on the tech stacks.
